git/git.go

```diff
@@ -0,0 +1,145 @@
+/*
+Copyright 2022 The Flux authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package git
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"io"
+	"strings"
+	"time"
+
+	"github.com/ProtonMail/go-crypto/openpgp"
+)
+
+type Hash []byte
+
+// String returns the SHA1 Hash as a string.
+func (h Hash) String() string {
+	return string(h)
+}
```

Comment:
With an eye on SHA2 support being available in `git`. I think it would be good to at least prepare this for the future by e.g. making the `Hash` a separate generic type, and the current `Hash []byte` a `SHA1 [20]byte`.

Or as other option: to remove the indications of it being strictly a SHA1 hash.

Reply:
It is hard to say what this would look like as the new hash function support seems to come as part of Git protocol V3.

The goal number 3 of the [hash function transition](https://git-scm.com/docs/hash-function-transition/) states that:
Users can use SHA-1 and SHA-256 identifiers for objects interchangeably.

Suggesting that an object could be referred by both, meaning that we could end up having both.

On top of that, the document suggests the use of SHA3-256 instead of SHA-256 (from the SHA2 family). Both of which would result in the same output length, but from the same input they yield different outputs.

All the above, makes me think that the second option - removing indications of it being stricly SHA1 - may be a first good step.